source/dicom/encaps.py

```diff
@@ -43,7 +41,18 @@ def defragment_data(data):
         seq.append(item)
 
     # XXX should
-    return "".join(seq)
+    return seq
+
+
+def defragment_data(data):
+    """Read encapsulated data and return one continuous string
+
+    data -- string of encapsulated data, typically dataset.PixelData
+    Return all fragments concatenated together as a byte string
+
+    If PixelData has multiple frames, then should separate out before calling this routine.
+    """
+    return "".join(decode_data_sequence(data))
```

Comment:
This refactors defragment_data into two methods, decode_data_sequence and the original defragment_data.  defragment_data maintains the original API and decode_data_sequence is used in dataset.py to decode multiframe compressed pixel data.